pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -446,13 +442,21 @@ public void publishMessage(ByteBuf headersAndPayload, PublishContext publishCont
     }
 
     public void updateSubscribeRateLimiter() {
-        SubscribeRate subscribeRate = this.getSubscribeRate();
+        SubscribeRate subscribeRate = getSubscribeRate();
         if (isSubscribeRateEnabled(subscribeRate)) {
-            subscribeRateLimiter = Optional.of(subscribeRateLimiter.orElse(new SubscribeRateLimiter(this)));
+            if (subscribeRateLimiter.isPresent()) {
+                if (!subscribeRateLimiter.get().getSubscribeRate().equals(subscribeRate)) {
```

Comment:
It might make sense to move this optimization into the `onSubscribeRateUpdate` method itself. The method could return early if the rate hasn't changed.

Reply:
Make sense. thx, PATL again @michaeljmarshall